mlflow/java/client/src/main/java/org/mlflow/client/ApiClient.java

```diff
@@ -0,0 +1,140 @@
+package org.mlflow.client;
+
+import org.apache.http.client.utils.URIBuilder;
+
+import org.mlflow.api.proto.Service.*;
+import org.mlflow.client.creds.BasicMlflowHostCreds;
+import org.mlflow.client.creds.MlflowHostCredsProvider;
+import org.mlflow.client.objects.FromProtobufMapper;
+import org.mlflow.client.objects.ToProtobufMapper;
+
+import java.net.URI;
+import java.util.List;
+import java.util.Optional;
+
+/**
+ * Client to an MLflow Tracking Sever.
+ */
+public class ApiClient {
```

Comment:
I think MlflowClient would be a better name for it. Also, should it love in `org.mlflow.client` or just `org.mlflow`? Since this is the main thing people will use, I'd put it in the top level package. Finally, if we call this client, we'll probably want to rename the Python one to client eventually too (doesn't need to be right now).